--- a/steamos-manager/src/session.rs	2026-03-07 06:59:16.292620248 +0000
+++ b/steamos-manager/src/session.rs	2026-03-07 06:59:21.442629761 +0000
@@ -6,8 +6,6 @@
  * SPDX-License-Identifier: MIT
  */
 
-#[cfg(test)]
-use anyhow::anyhow;
 use anyhow::{Result, ensure};
 use serde::{Deserialize, Serialize};
 use std::ffi::OsStr;
@@ -30,7 +28,6 @@
 use crate::{Service, path};
 
 const CONFIG_PREFIX: &str = "/etc/sddm.conf.d";
-const SESSION_CHECK_PATH: &str = "steamos.conf";
 const CONFIG_PATH: &str = "zz-steamos-autologin.conf";
 const TEMPORARY_CONFIG_PATH: &str = "zzt-steamos-temp-login.conf";
 
@@ -91,18 +88,6 @@
     LoginModeChanged(LoginMode),
 }
 
-pub(crate) async fn is_session_managed() -> Result<bool> {
-    Ok(try_exists(path(CONFIG_PREFIX).join(SESSION_CHECK_PATH)).await?)
-}
-
-#[cfg(test)]
-pub(crate) async fn make_managed() -> Result<()> {
-    let check_path = path(CONFIG_PREFIX).join(SESSION_CHECK_PATH);
-    create_dir_all(check_path.parent().ok_or(anyhow!("Couldn't make dir"))?).await?;
-    write(check_path, "").await?;
-    Ok(())
-}
-
 fn is_valid_desktop_session_name(path: &str) -> bool {
     if path.starts_with('.') {
         return false;
