--- a/steamos-manager/src/wifi.rs	2026-03-07 07:10:25.256850756 +0000
+++ b/steamos-manager/src/wifi.rs	2026-03-07 07:10:25.269850780 +0000
@@ -33,7 +33,7 @@
 
 const OVERRIDE_CONTENTS: &str = "[Service]
 ExecStart=
-ExecStart=/usr/lib/iwd/iwd -d
+ExecStart=/usr/libexec/iwd/iwd -d
 ";
 const OVERRIDE_FOLDER: &str = "/etc/systemd/system/iwd.service.d";
 const OVERRIDE_PATH: &str = "/etc/systemd/system/iwd.service.d/99-valve-override.conf";
