--- a/steamos-manager/src/inputplumber.rs	2026-03-07 07:06:09.243381689 +0000
+++ b/steamos-manager/src/inputplumber.rs	2026-03-07 07:06:09.277381751 +0000
@@ -39,11 +39,13 @@
 }
 
 #[derive(Clone, Debug)]
+#[allow(dead_code)]
 pub struct DeckService {
     connection: Connection,
     composite_device_iface_name: OwnedInterfaceName,
 }
 
+#[allow(dead_code)]
 impl DeckService {
     pub fn init(connection: Connection) -> DeckService {
         DeckService {
