diff --git a/Makefile b/Makefile
index 1a33cf2..60e51f1 100644
--- a/Makefile
+++ b/Makefile
@@ -43,4 +43,3 @@ install: target/release/steamos-manager target/release/steamosctl
 	install -m644 "data/user/com.steampowered.SteamOSManager1.service" "$(DESTDIR)/usr/share/dbus-1/services/"
 	install -m644 "data/user/steamos-manager.service" "$(DESTDIR)/usr/lib/systemd/user/"
 	install -m644 "data/user/steamos-manager-session-cleanup.service" "$(DESTDIR)/usr/lib/systemd/user/"
-	install -m644 "data/user/orca.service" "$(DESTDIR)/usr/lib/systemd/user/"
