diff --git a/Makefile b/Makefile
index 7bc72d5..99f5460 100644
--- a/Makefile
+++ b/Makefile
@@ -44,4 +44,3 @@ install: target/release/steamos-manager target/release/steamosctl
 	install -m644 "data/user/steamos-manager.service" "$(DESTDIR)/usr/lib/systemd/user/"
 	install -m644 "data/user/steamos-manager-session-cleanup.service" "$(DESTDIR)/usr/lib/systemd/user/"
 	install -m644 "data/user/steamos-manager-configure-cecd.service" "$(DESTDIR)/usr/lib/systemd/user/"
-	install -m644 "data/user/orca.service" "$(DESTDIR)/usr/lib/systemd/user/"
